src/basic/unit-name.c

```diff
@@ -821,3 +835,11 @@ bool unit_name_prefix_equal(const char *a, const char *b) {
 
         return memcmp_nn(a, p - a, b, q - b) == 0;
 }
+
+const UnitNameVTable default_unit_name_vtable = {
+        .max_name = 256,
+};
+
+const UnitNameVTable long_unit_name_vtable = {
+        .max_name = UNIT_NAME_LIMIT,
+};
```

Comment:
Hmm, why define a "vtable" for this? It seems unnecessarily complicated.

```C
static inline size_t unit_name_max_length(UnitType type) {
    assert(type >= 0);
    if (IN_SET(type, UNIT_DEVICE, UNIT_MOUNT, UNIT_AUTOMOUNT, UNIT_SWAP))
            /* Add a comment here why we allow longer names for those types… */
            return UNIT_NAME_MAX_LONG;
    return UNIT_NAME_MAX_DEFAULT;
}
```

(And I think it's necessary to adjust the macro names, because it'd be misleading to have UNIT_NAME_MAX when we allow longer unit names.)

Reply:
Initially I wanted this to become part of already existing vtable structs. But that would create "bad" dependency between translation units (src/basic/unit-name.c would then depend on stuff from src/core/) I had to separate it to the own struct. Since maximum unit name length is now becoming unit type specific property I sticked to table approach.

Anyway, since we only have single new property the function is indeed simpler so let's do it that way.